src/UserAction/StepExecutorTrait.php

```diff
@@ -188,11 +215,11 @@ protected function doFields(View $page): void
 
         if (!$form->hookHasCallbacks(Form::HOOK_SUBMIT)) {
             $form->onSubmit(function (Form $form) {
-                // collect fields.
-                $form_fields = $form->model->get();
-                foreach ($this->action->fields as $field) {
-                    $this->actionData['fields'][$field] = $form_fields[$field];
-                }
+                // collect fields define in Model\UserAction.
+                $fields = array_filter($form->model->get(), function ($field) {
```

Comment:
`array_intersect_key($form->model->get(), array_flip($this->action->fields))`

Reply:
what is wrong with array_filter?

Reply:
a) no need to fn if built in function like `array_intersect_key` does the same
b) we should not materialize all fields and filter afterwards, we should materialize only the needed ones (as some might me lazy loaded in the future)

Reply:
update with array_intersect